src/Java.Interop/Java.Interop/JniRuntime.JniTypeManager.cs

```diff
@@ -211,13 +211,15 @@ IEnumerable<Type> CreateGetTypesForSimpleReferenceEnumerator (string jniSimpleRe
 			}
 
 			public virtual void RegisterNativeMembers (JniType nativeClass, Type type, string methods)
+			{
+				TryRegisterNativeMembers (nativeClass, type, methods);
+			}
+
+			protected bool TryRegisterNativeMembers (JniType nativeClass, Type type, string methods)
 			{
 				AssertValid ();
 
-				if (!TryLoadJniMarshalMethods (nativeClass, type, methods) &&
-						!TryRegisterNativeMembers (nativeClass, type, methods)) {
-					throw new NotSupportedException ($"Could not register Java.class={nativeClass.Name} Managed.type={type.FullName}");
-				}
+				return TryLoadJniMarshalMethods (nativeClass, type, methods) || TryRegisterNativeMembers (nativeClass, type, methods);
```

Comment:
This is recursive & stack overflow.

You want `TryRegisterNativeMembers (nativeClass, type, methods, registerMethod:null)`.

Reply:
ops :-) Will fix promptly.